src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,6 +88,11 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+        File storage = path.toFile();
+        if (!storage.isFile()) {
+            throw new StorageOperationException("File deleted during operation: " + path);
```

Comment:
Where are you catching this exception?
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.

You can either(1) warn and terminate or (2) warn and create the file.